src/generator-nullability-info-context.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+
+#nullable enable
+
+namespace bgen {
+
+	// from: https://github.com/dotnet/roslyn/blob/main/docs/features/nullable-metadata.md
+	public enum NullableFlag : byte {
+		Oblivious = 0,
+		NotAnnotated = 1,
+		Annotated = 2,
+	}
+
+	public class NullabilityInfo {
+		public bool IsNullable { get; set; }
+		public Type? Type { get; set; }
+		public NullabilityInfo? ElementType { get; set; }
+		public NullabilityInfo []? GenericTypeArguments { get; set; }
+	}
+
+	// helper class that allows to check the custom attrs in a method,property or field
+	// in order for the generator to support ? in the bindings definition. This class should be
+	// replaced by NullabilityInfoContext from the dotnet6 in the future. The API can be maintained (hopefully).
+	public static class GeneratorNullabilityInfoContext {
```

Comment:
You can do something like this:

```cs
#if !NET
namespace System.Reflection {
    public static class NullabilityInfoContext {
        // same API as https://learn.microsoft.com/en-us/dotnet/api/system.reflection.nullabilityinfocontext.create?view=net-7.0
    }
}
#endif // !NET
```

and then you can run the tests on both .NET and legacy Xamarin, and if they all pass, you'll know both the tests and the legacy-only implementation are correct (because the .NET tests will run with NullabilityInfoContext from the BCL).

This way all the code in the generator will be identical between legacy Xamarin and .NET.

Reply:
That was my plan hehe, I was just looking at getting this to work, but did it in this PR.